src/overlays/actors/ovl_En_Zod/z_en_zod.c

```diff
@@ -28,62 +60,592 @@ const ActorInit En_Zod_InitVars = {
+    msgState = Message_GetState(&play->msgCtx);
+    if (msgState != TEXT_STATE_CLOSING) {
+        if ((msgState == 5) && Message_ShouldAdvance(play)) {
+            func_801477B4(play);
+            this->actionFunc = func_80BAFB84;
+            EnZod_ChangeAnimation(this, ENZOD_ANIM_PLAYING_LENTO, ANIMMODE_ONCE);
+        }
+    } else {
+        this->actionFunc = func_80BAFB84;
+        EnZod_ChangeAnimation(this, ENZOD_ANIM_PLAYING_LENTO, ANIMMODE_ONCE);
+    }
+}
+
+void func_80BAFB84(EnZod* this, PlayState* play) {
+    EnZod_UpdateAnimations(this);
+
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        func_80BAFA44(this, play);
+        this->actionFunc = func_80BAFADC;
+    } else if (EnZod_PlayerIsFacingTijo(this, play)) {
+        func_800B8614(&this->actor, play, 210.0f);
+    }
+}
+
+void EnZod_DoNothing(EnZod* this, PlayState* play) {
+}
+
+void EnZod_RehearsalCutsceneTransition(EnZod* this, PlayState* play) {
+    EnZod_UpdateAnimations(this);
+    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+        ActorCutscene_Start(this->actor.cutscene, &this->actor);
+        this->actor.cutscene = ActorCutscene_GetAdditionalCutscene(this->actor.cutscene);
+        if (this->actor.cutscene == -1) {
+            this->actionFunc = EnZod_DoNothing;
+            play->nextEntrance = play->setupExitList[ENZOD_GET_ENTRANCE_INDEX(&this->actor)];
+            play->transitionType = TRANS_TYPE_05;
+            play->transitionTrigger = TRANS_TRIGGER_START;
+            gSaveContext.save.weekEventReg[0x4E] &= 0xFE;
+        } else {
+            ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        }
+    } else {
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    }
+}
+
+void EnZod_StartSession(EnZod* this, PlayState* play) {
+    EnZod_UpdateAnimations(this);
+    if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && Message_ShouldAdvance(play)) {
+        func_801477B4(play);
+        EnZod_ChangeAnimation(this, ENZOD_ANIM_PLAYING_LENTO, ANIMMODE_ONCE);
+        this->actionFunc = EnZod_RehearsalCutsceneTransition;
+        ActorCutscene_Stop(this->actor.cutscene);
+        this->actor.cutscene = ActorCutscene_GetAdditionalCutscene(this->actor.cutscene);
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        gSaveContext.save.weekEventReg[0x4F] |= 1;
+        Audio_QueueSeqCmd(NA_BGM_INDIGO_GO_SESSION | 0x8000);
+    }
+}
+
+void func_80BAFDB4(EnZod* this, PlayState* play) {
+    EnZod_UpdateAnimations(this);
+    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+        ActorCutscene_Start(this->actor.cutscene, &this->actor);
+        func_800B7298(play, NULL, 0x44);
+        Message_StartTextbox(play, 0x103A, &this->actor); // One... two... three...
+        this->actionFunc = EnZod_StartSession;
+    } else {
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    }
+}
+
+void EnZod_UpdateFog(EnZod* this, PlayState* play) {
```

Comment:
I agree with what Fig mentioned in discord, I feel like we could name this more on what he is doing in the cutscene. Tbh, I would probably just leave it unnamed for now.